api_tests/nodes/views/test_node_list.py

```diff
@@ -346,6 +376,31 @@ def test_incorrect_filtering_field_not_logged_in(self):
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], "'notafield' is not a valid field for this endpoint.")
 
+    def test_filtering_on_root(self):
+        # Build up a family of nodes
+        node_structure = [5, 2, [1, 2], 3, [10, 2], 2, [4, [2, 2], 6]]
+        render_generations_from_node_structure_list(self.project_one, node_structure)
```

Comment:
On a slightly more serious note, my only real concern is if someone refactors the function that gathers all of the children into a flat list in such a way that, on some structure of node, will return a child node earlier in a list than the parent node. This test is, perhaps, a bit more aggressive than it needs to be on that front, but if that happens, it would be pretty terrible. Not disastrous, though; once the migration is done, the only likely scenario will be if someone is moving a tree structure from one node to another, and that *could* be fixed by hand if need be.